core/src/main/java/org/apache/iceberg/rest/auth/HttpRequestFacade.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.rest.auth;
+
+import com.google.errorprone.annotations.CanIgnoreReturnValue;
+import java.net.URI;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * A facade for an HTTP request that allows reading and modifying the request. This is used by
+ * {@link AuthManager} implementations, e.g. to set authentication headers on the request.
+ */
+public interface HttpRequestFacade {
+
+  /** Returns the full request URI of this request. */
+  URI getUri();
```

Comment:
I think this interface should conform to existing Iceberg conventions and should not expose `get` in method names.